### PR TITLE
Display error messages for user

### PR DESCRIPTION
**Pull request by arisachue**
- Add in Alert messages for user when error is present
- Fix recipe page bug (described below)
1. when the user first logs in, they had to first go to the pantry page and then go to the recipe page since the pantry page was updating the current user info. fix: update the current user info in the recipe page as well
2. when the user has zero product items, recipe doesn't display anything. fix: have a separate api request call that displays standard popular recipes

Note: I maxed out on my recipes API request limit, so I haven't been able to test the recipes error handling

Bugs I haven't been able to fix yet:
- Every page refresh, the website renders twice?
- Every page refresh, the current user login info remains the same, but the user info from the database becomes undefined

### CODE REVIEW

File: src/components/RecipeCard/RecipeCard.jsx [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
 
 export default function RecipeCard(props) {
+  const recipeInfoUrl = "http://localhost:3001/apirecipeinfo/"
+  const [recipeModelShow, setRecipeModalShow] = useState(false)
+  const [recipeInfo, setRecipeInfo] = useState({})
+  const [error, setError] = useState("")
+
+  // onclick function for each recipe card
+  const handleRecipeCardClick = async (showStatus, recipeId) => {
+    setError("")
```
Comment (reddysindhu1) [inline src/components/RecipeCard/RecipeCard.jsx]: Why are you setError("") here? Shouldn't it already be "" since that's what you originally set it to?

File: src/components/Recipes/Recipes.jsx [hunk 2]
```diff
@@ -1,16 +1,74 @@
...
+        setRecipes(data)
+      } catch (err) {
+        setError(err.message)
+      }
+    }
+
+    if (userProducts.length === 0) {
+      fetchStdRecipes()
+    }
+  }, [])
```
Comment (reddysindhu1) [inline src/components/Recipes/Recipes.jsx]: Your undefined issue _might_ be because of this useEffect. Maybe try passing in [currentUser.uid] as the second argument. 
According to this docs, https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects, passing an empty array will only call the useEffect on mount and unmount so maybe when you rerender you're not actually calling this effect and it's defaulting to the initial {} for userInfo.
image

**Review by reddysindhu1 — CHANGES_REQUESTED** [review]

File: src/components/Pantry/Pantry.jsx [hunk 3]
```diff
@@ -27,29 +25,42 @@ export default function Pantry(props) {
...
   }
   const [productForm, setProductForm] = useState(basicProductForm)
-  const [userProducts, setUserProducts] = useState(userInfo.data.products)
+  const [userProducts, setUserProducts] = useState([])
+  const [error, setError] = useState("")
+
+  // update user data once page loads
+  useEffect(() => {
+    if (!props.isLoading) {
+      var curuser = props.users.find(u => u.uid === currentUser.uid)
```
Comment (reddysindhu1) [inline src/components/Pantry/Pantry.jsx]: Nit: curUser instead of curuser so it's standardized.

File: src/components/RecipeModal/RecipeModal.jsx [hunk 4]
```diff
@@ -8,6 +8,23 @@ export default function RecipeModal(props) {
         return null
     }
     const { recipeInfo, ...modalProps } = props
+    var validMins = true
```
Comment (reddysindhu1) [inline src/components/RecipeModal/RecipeModal.jsx]: You can condense validMins, validServings, validIngred, and validSteps.
const validMins = !(recipeInfo.readInMinutes == undefined)

File: src/components/RecipeModal/RecipeModal.jsx [hunk 5]
```diff
@@ -19,32 +36,44 @@ export default function RecipeModal(props) {
           <Container>
             {/* first row - recipe overview */}
             <Row className="modal-overview">
-                <Col>
-                Total Time: {recipeInfo.readyInMinutes} mins
-                </Col>
-                <Col>
-                Servings: {recipeInfo.servings}
-                </Col>
+              {validMins
```
Comment (reddysindhu1) [inline src/components/RecipeModal/RecipeModal.jsx]: Since we're not displaying anything if validMins isn't true we can do 
{validMins && (<Col> Total Time: {recipeInfo.readyInMinutes} mins </Col>)}

File: src/components/RecipeModal/RecipeModal.jsx [hunk 6]
```diff
@@ -19,32 +36,44 @@ export default function RecipeModal(props) {
...
-                Total Time: {recipeInfo.readyInMinutes} mins
-                </Col>
-                <Col>
-                Servings: {recipeInfo.servings}
-                </Col>
+              {validMins 
+                ? <Col> Total Time: {recipeInfo.readyInMinutes} mins </Col>
+                : null
+              }
+              {validServings
```
Comment (reddysindhu1) [inline src/components/RecipeModal/RecipeModal.jsx]: Ditto above comment

File: src/components/Recipes/Recipes.jsx [hunk 7]
```diff
@@ -1,16 +1,73 @@
...
+  const listUserRecipeUrl = "http://localhost:3001/apiuserrecipes/"
+  const listStdRecipeUrl = "http://localhost:3001/apistdrecipes"
+  const [recipes, setRecipes] = useState([])
+  const [userProducts, setUserProducts] = useState([])
+  const [error, setError] = useState("")
+
+  // update user data once page loads
+  useEffect(() => {
+    if (!props.isLoading) {
+      var curuser = props.users.find(u => u.uid === currentUser.uid)
```
Comment (reddysindhu1) [inline src/components/Recipes/Recipes.jsx]: Nit: curUser instead of curuser

File: express-server/app.js [hunk 8]
```diff
@@ -36,4 +36,19 @@ app.get('/apirecipeinfo/:recipeid', async (request, response) => {
...
+        sort: "random",
+        addRecipeInformation: true,
+        instructionsRequired: true,
+        fillIngredients: true
+    }
+    let recipes_url = `https://api.spoonacular.com/recipes/complexSearch`;
+    let { data } = await axios(recipes_url, { params })
+    response.json(data.results);
+});
+
```
Comment (mjquigley) [inline express-server/app.js:53]: This doesn't seem related to error handling, it should probably be in it's own PR

File: src/components/Pantry/Pantry.jsx [hunk 9]
```diff
@@ -27,29 +25,42 @@ export default function Pantry(props) {
...
+      setUserProducts(curuser.data.products)
+    }
+  }, [props.isLoading])
+  
 
   // update products state
   useEffect(() => {
-    props.setProducts(userProducts)
+    console.log("updating database")
```
Comment (mjquigley) [inline src/components/Pantry/Pantry.jsx]: Remove this log or if it's useful for now, add a TODO to remove it later.

File: src/components/Pantry/Pantry.jsx [hunk 10]
```diff
@@ -74,22 +84,27 @@ export default function Pantry(props) {
     let itemIndex = userProducts.findIndex(item => item.name === itemName)
 
     let newProducts = [...userProducts]
-
+    setError("")
```
Comment (mjquigley) [inline src/components/Pantry/Pantry.jsx]: nit: a helper function like `clearError()` might make the intent here more clear by providing a better name.

File: src/components/RecipeModal/RecipeModal.jsx (hunk 4 above)
Comment (mjquigley) [reply src/components/RecipeModal/RecipeModal.jsx]: Agreed. This would make this much easier to read.

**Comment by mjquigley** [comment]
I think this PR could have been broken up into several smaller ones. This was a lot to review at once which makes it harder to find bugs. 

Another option for making these PRs easier to review is to at least split  the changes up into multiple commits. With each commit being a single small change. The PR review tool lets you review PRs 1 commit at a time. So that's another option for improving your PR workflow. 

A good rule of thumb I follow when submitting a PR is: Does my PR description contain the word "and" or a list of changes? If so, each of those changes could usually be it's own PR.

**Review by mjquigley — APPROVED** [review]
